openstack/client.go

```diff
@@ -414,3 +414,9 @@ func NewKeyManagerV1(client *gophercloud.ProviderClient, eo gophercloud.Endpoint
 	sc.ResourceBase = sc.Endpoint + "v1/"
 	return sc, err
 }
+
+// NewContainerInfraV1 creates a ServiceClient that may be used with the v1 container infra management
+// package.
+func NewContainerInfraV1(client *gophercloud.ProviderClient, eo gophercloud.EndpointOpts) (*gophercloud.ServiceClient, error) {
+	return initClientOpts(client, eo, "container-infra")
```

Comment:
In the `urls.go` file you are trying to determine if `/v1` is included in the URL. There isn't precedence for that which would make this a very rare case. I recommend adding `v1` here like how it's done in the other clients in this file.

Reply:
Magnum includes version in its endpoint, so adding it will make sc.ResourceBase have double "v1". But testing doesn't include "v1" for its endpoint. Will look at it more to see what needs to be done.

Reply:
ah - I understand.

In this case, I recommend setting `/v1` in the ServiceClient and removing it from `urls.go`. The `testhelper` functions and testclient will work with this configuration.